### DISCUSSION TITLE
Custom page is 404.

### DISCUSSION THREAD

**Discussion by dogeow** [filamentphp/filament#6197] [Help] [answered]
Thank you for your time.

## Start

```shell
php artisan make:filament-page LotteryTask --resource=TaskResource --type=custom
```

## Resource

Add `lottery` path

```php
 public static function getPages(): array
    {
        return [
            'index' => Pages\ListTasks::route('/'),
            'create' => Pages\CreateTask::route('/create'),
            'edit' => Pages\EditTask::route('/{record}/edit'),
            'view' => Pages\ViewTask::route('/{record}'),
            'lottery' => Pages\LotteryTask::route('/lottery'),
        ];
    }
```

## Route

```
php artisan optimize:clear
```

![image](https://user-images.githubusercontent.com/7204604/231814679-4551b364-ebb7-4581-a1cd-c1e4e845c67b.png)

## Result

Other pages is ok, such `/admin/tasks/*` pages. But ...

![image](https://user-images.githubusercontent.com/7204604/231815381-05766324-3460-4a39-aa11-787828b14bfd.png)

**Comment by danharrin** [comment, marked as answer]
But the `view` after `lottery` because it is a wildcard route that will be caught first by the router

**Reply by tsalim13** [reply]
I am new to filament and I came across this problem, can you explain it to me more clearly? I did not understand well

**Reply by lucacastelnuovo** [reply]
The route definition `admin/tasks/{record}` can also catch `admin/tasks/lottery`. 
This behaviour is not a feature/bug of Filament but also exists in Laravel routing (see: https://laravel.com/docs/10.x/routing#route-parameters)

In `web.php` you can constrain what strings the route parameter catches (see: https://laravel.com/docs/10.x/routing#parameters-regular-expression-constraints), using these constraints you could ensure `lottery` is not caught in the parameter.
I do not believe that this is possible in Filament routing.

You should change the lottery route or edit and create routes.

The following could work but is not extremely pretty (it feels kinda off in my opinion).
```php
public static function getPages(): array
{
    return [
        'index' => Pages\ListTasks::route('/'),
        'create' => Pages\CreateTask::route('/create'),
        'edit' => Pages\EditTask::route('/form/{record}/edit'),
        'view' => Pages\ViewTask::route('/form/{record}'),
        'lottery' => Pages\LotteryTask::route('/lottery'),
    ];
}
```

**Reply by lucacastelnuovo** [reply]
I haven't tested it but maybe the following will work:

```php
public static function getPages(): array
{
    return [
        'index' => Pages\ListTasks::route('/'),
        'create' => Pages\CreateTask::route('/create'),
        'lottery' => Pages\LotteryTask::route('/lottery'),
        'edit' => Pages\EditTask::route('/{record}/edit'),
        'view' => Pages\ViewTask::route('/{record}'),
    ];
}
```

I thought of this because obviously `create` also does not get caught in the `{record}` parameter.

Let me know if it works :)

**Reply by tsalim13** [reply]
it work thank you

**Comment by ahmadrio** [comment]
> I haven't tested it but maybe the following will work:
> 
> ```
> public static function getPages(): array
> {
>     return [
>         'index' => Pages\ListTasks::route('/'),
>         'create' => Pages\CreateTask::route('/create'),
>         'lottery' => Pages\LotteryTask::route('/lottery'),
>         'edit' => Pages\EditTask::route('/{record}/edit'),
>         'view' => Pages\ViewTask::route('/{record}'),
>     ];
> }
> ```
> 
> I thought of this because obviously `create` also does not get caught in the `{record}` parameter.
> 
> Let me know if it works :)

thanks it works @lucacastelnuovo 🙏
